Disable sevice reload using "docker:reload" pillar

vin01 opened the pull request:
Some docker configuration options like userns-remap, metrics-addr require a restart for new configuration to take effect, with a reload they are silently required. Having an interactive pillar which can be used to force restart can be helpful in these cases.

chr4 commented:
Maybe we can keep it consistent with [postgresql](https://github.com/chr4/salt-postgresql/blob/main/init.sls#L9)

```yaml
reload: {{ pillar['postgresql']['reload']|default('true') }}
```

vin01 committed:
Allow disabling sevice reload using "docker:reload" pillar

Some docker configuration options like userns-remap, metrics-addr require a restart for new configuration to take effect, with a reload they are silently required. Having an interactive pillar which can be used to force restart can be helpful in these cases.

vin01 commented:
hmm, I was looking for a shorthand, but this should also be fine, updated!

```
salt-call state.apply pillar='{"docker":{"reload":False}}'
```
